metrics_ci_integration/lib/jenkins/model/jenkins_job.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:ci_integration/jenkins/model/jenkins_building_job.dart';
+import 'package:ci_integration/jenkins/model/jenkins_multi_branch_job.dart';
+import 'package:equatable/equatable.dart';
+import 'package:meta/meta.dart';
+
+/// A class representing a Jenkins job.
+///
+/// Instances contain the general information about a Jenkins job, while
+/// [JenkinsBuildingJob] and [JenkinsMultiBranchJob] specifies this information.
+class JenkinsJob extends Equatable {
+  final String name;
+  final String fullName;
+  final String url;
+
+  @override
+  List<Object> get props => [name, fullName, url];
+
+  const JenkinsJob({
+    @required this.name,
+    this.fullName,
+    this.url,
+  });
```

Comment:
Should we assert that name is not null?

Reply:
Not sure we should require this at all since the `fullName` property sometimes more useful and we use a `url` property more often.